### PR TITLE
Improved error when adding project without admin rights.

### PR DESCRIPTION
**Pull request by alexfernandez**
When trying to add a project where you don't have admin rights, right now the error message is:

    Error creating project for repo owner/project:
    {"results":[],"status":"error","errors":[{"code":500,
    "reason":"Failed to setup repo: Bad status code: 404"}]}

Now should show:

    "reason":"Failed to setup repo: Cannot create webhooks; are you sure you have admin rights?"}]}

Also added a known issue to the `README.md`.

Should solve issue https://github.com/Strider-CD/strider-github/issues/60.

### CODE REVIEW

File: lib/api.js [hunk 1]
```diff
@@ -50,8 +50,14 @@ function createHooks(reponame, url, secret, token, callback) {
     .end(function (err, res) {
       if (err) return callback(err);
 
+      var badStatusErr;
+      if (res.statusCode === 404) {
+        badStatusErr = new Error('Cannot create webhooks; are you sure you have admin rights?');
+        badStatusErr.statusCode = res.statusCode;
+        return callback(badStatusErr);
+      }
       if (res.statusCode !== 201) {
```
Comment (knownasilya) [inline lib/api.js]: Probably worth doing `else if (...)` here.

**Comment by alexfernandez** [comment]
Changed in commit a93098f.

**Comment by knownasilya** [comment]
Published as 2.2.1

**Comment by alexfernandez** [comment]
Great, thanks! This way other clueless newbies like myself will have it easier :)

**Comment by knownasilya** [comment]
@alexfernandez really appreciate it. This project can only stay alive because of people like you, who contribute back.

**Comment by oliversalzburg** [comment]
I'm not sure if this would work, but couldn't we include `url` in the error message? 

As in:
```js
badStatusErr = new Error('Cannot create webhooks; are you sure you have admin rights?\nFeel free to manually create a webhook for ' + url);
```

**Comment by knownasilya** [comment]
@oliversalzburg good idea, will add it.

**Comment by oliversalzburg** [comment]
@knownasilya I'm not sure if the hook will actually work if the registration process failed. As in, maybe it's a different URL for each registration attempt. I don't believe this to be the case, but I'm not sure.

Also, I'm not sure the the way the error is displayed allows for copy&paste of the URL.

If neither is a problem, great :)

**Comment by knownasilya** [comment]
I guess we shall see :+1:

**Pull request merged**